### PR TITLE
feat(storage): replace risingwave lru cache with foyer lru cache for lsm cache

### PR DESCRIPTION
**Pull request by MrCroxx**
Signed-off-by: MrCroxx <mrcroxx@outlook.com>I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Bump foyer to 0.6

Replace the current LRU cache implementation with foyer in-memory cache.

The in-memory cache of foyer will simplify the development and experiences of cache eviction algorithms. And will make the eviction algorithm pluggable to make configuration easier.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by stdrc** [comment]
> Not ready yet. Just trigger e2e.

You can add `ci/...` labels to trigger ci workflows on draft pr.

File: src/storage/src/hummock/sstable_store.rs [hunk 1]
```diff
@@ -612,50 +665,43 @@ impl SstableStore {
...
+            let meta_path = self.get_sst_data_path(object_id);
+            let stats_ptr = stats.remote_io_time.clone();
+            let range = sst.meta_offset as usize..sst.file_size as usize;
+            async move {
+                if let Some(sst) = meta_file_cache
+                    .lookup(&object_id)
+                    .await
+                    .map_err(HummockError::file_cache)?
+                {
+                    // TODO(MrCroxx): Make meta cache receives Arc<Sstable> to reduce copy?
```
Comment (Li0k) [inline src/storage/src/hummock/sstable_store.rs:680]: Is this possible? Sstable contains BlockMeta and Filter, its size depends on block_count and dist key count

**Review by Li0k — APPROVED** [review]
LGTM!!! Thanks for the effort !!!

File: src/storage/src/hummock/sstable_store.rs (hunk 1 above)
Comment (MrCroxx) [reply src/storage/src/hummock/sstable_store.rs:680]: It's supposed to be possible if it is immutable. 🤔

**Review by hzxa21 — APPROVED** [review]
LSTM! Thanks for the efforts

**Pull request merged**